pkg/network/multus/status_test.go

```diff
@@ -33,72 +32,133 @@ import (
+				`{"name":"meganet","interface":"pod7e0055a6880","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
+				`]`
 
-		Expect(result).To(Equal(expectedResult))
+			annotations := map[string]string{networkv1.NetworkStatusAnnot: multusNetworkStatusWithPrimaryAndSecondaryNets}
+
+			expectedResult := []networkv1.NetworkStatus{
+				{
+					Name:    "k8s-pod-network",
+					IPs:     []string{"10.244.196.146", "fd10:244::c491"},
+					Default: true,
+					DNS:     networkv1.DNS{},
+				},
+				{
+					Name:      "meganet",
+					Interface: "pod7e0055a6880",
+					Mac:       "8a:37:d9:e7:0f:18",
+					Default:   false,
+					DNS:       networkv1.DNS{},
+				},
+			}
+
+			Expect(multus.NetworkStatusesFromPod(newStubPod(annotations))).To(Equal(expectedResult))
+		})
 	})
 
-	It("Should return a map of pod interface name to network status when interface name is ordinal", func() {
+	Context("LookupPodPrimaryIfaceName", func() {
 		const (
-			multusOrdinalNetworksAnnotation                       = `[{"name":"meganet","namespace":"default","interface":"net1"}]`
-			multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets = `[` +
-				`{"name":"k8s-pod-network","ips":["10.244.196.146","fd10:244::c491"],"default":true,"dns":{}},` +
-				`{"name":"meganet","interface":"net1","mac":"8a:37:d9:e7:0f:18","dns":{}}` +
-				`]`
+			defaultPrimaryPodIfaceName = "eth0"
+			customPrimaryPodIfaceName  = "custom-iface"
 		)
 
-		annotations := map[string]string{
-			networkv1.NetworkAttachmentAnnot: multusOrdinalNetworksAnnotation,
-			networkv1.NetworkStatusAnnot:     multusNetworkStatusWithPrimaryAndOrdinalSecondaryNets,
-		}
-
-		result := multus.NonDefaultNetworkStatusIndexedByIfaceName(newStubPod(annotations))
-
-		expectedResult := map[string]networkv1.NetworkStatus{
-			"net1": {
-				Name:      "meganet",
-				Interface: "net1",
-				Mac:       "8a:37:d9:e7:0f:18",
-				Default:   false,
-			},
-		}
+		DescribeTable("should return empty string", func(networkStatuses []networkv1.NetworkStatus) {
+			Expect(multus.LookupPodPrimaryIfaceName(networkStatuses)).To(BeEmpty())
+		},
+			Entry("when network status is nil", nil),
+			Entry("when network status is empty", []networkv1.NetworkStatus{}),
+			Entry("when network status does not report interface name",
+				[]networkv1.NetworkStatus{
+					{Name: "k8s-pod-network", Default: true},
+				},
+			),
+		)
 
-		Expect(result).To(Equal(expectedResult))
+		DescribeTable("Should return the primary pod interface name", func(networkStatuses []networkv1.NetworkStatus, expectedResult string) {
+			Expect(multus.LookupPodPrimaryIfaceName(networkStatuses)).To(Equal(expectedResult))
+		},
+			Entry("When the primary pod interface name is reported and its the default value",
+				[]networkv1.NetworkStatus{
+					{Name: "k8s-pod-network", Interface: defaultPrimaryPodIfaceName, Default: true},
+					{Name: "some-net", Interface: "pod123456", Default: false},
```

Comment:
nit: no need to define the `Default` attribute, right ? 

It's opinionated, your call.

Reply:
Could you please elaborate why not?

Reply:
@maiqueb isn't that the idea per [proposed design] ?(https://github.com/kubevirt/community/blob/2bebe5193a88ebdafcc3c2f1b3432d9c3ac726d0/design-proposals/dynamic-primary-pod-nic-name.md?plain=1#L192)

Reply:
> I guess you mean because it is the default.

yes, because booleans default to false.

> I prefer to explicitly state the Default field's value - as it directly affects the logic.

all good, I honestly don't care.